clients/cli/README.md

```diff
@@ -0,0 +1,141 @@
+<!--
+  Licensed to the Apache Software Foundation (ASF) under one
+  or more contributor license agreements.  See the NOTICE file
+  distributed with this work for additional information
+  regarding copyright ownership.  The ASF licenses this file
+  to you under the Apache License, Version 2.0 (the
+  "License"); you may not use this file except in compliance
+  with the License.  You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing,
+  software distributed under the License is distributed on an
+  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+  KIND, either express or implied.  See the License for the
+  specific language governing permissions and limitations
+  under the License.
+-->
+
+# Apache Gravitino CLI
+
+Apache Gravitino CLI is a command-line tool that interacts with the Gravitino server to manage and query entities like metalakes, catalogs, schemas, and tables. The tool provides options for listing information about Gravitino entities and in future versions support creating, deleting, and updating these entities.
+
+## Table of Contents
+
+- [Features](#features)
+- [Installation](#installation)
+- [Usage](#usage)
+- [Commands](#commands)
+- [Running Tests](#running-tests)
+- [Contributing](#contributing)
+- [License](#license)
+
+## Features
+
+- Retrieve server version
+- Provide help on usage
+- Manage Gravitino entities such as Metalakes, Catalogs, Schemas, and Tables
+- List details about Graviotino entities
+
+## Installation
+
+### Prerequisites
+
+Before you can build and run this project, it is suggested you have the following installed:
+
+- Java 11 or higher
+
+### Build the Project
+
+1. Clone the entire Gravitino repository:
+
+    ```bash
+    git clone https://github.com/apache/gravitino
+    ```
+
+2. Build the CLI sub-project using Gradle:
+
+    ```bash
+    ./gradlew :clients:cli:build
+    ```
+3. Create an alias:
+
+    ```bash
+    alias gcli='java -jar clients/cli/build/libs/gravitino-cli-0.7.0-incubating-SNAPSHOT.jar'
+    ```
+3. Test the command:
+    ```bash
+    gcli --help
+    ```
+
+## Usage
+
+To run the Gravitino CLI, use the following command structure:
+
+```bash
+usage: gcli [metalake|catalog|schema|table] [list|details|create|delete|update] [options]
+Options
+ -c,--catalog <arg>    catalog name
+ -C,--create           create an entity
```

Comment:
As mentioned in the docs, there are several ways to format the command depending on your preference. So you can use either `--create` or `create` to create something.

Reply:
I don't think it's necessary to provide several formats of commands, as this will increase the burden of use and is not consistent with the usage you mentioned above:
`usage: gcli [metalake|catalog|schema|table] [list|details|create|delete|update] [options]`

Reply:
Several people who reviewed the design documentation including @jerryshao expressed a preference for this style of usage, perhaps check with them first?

Reply:
No, we should only deliver one consistent way. We can use like "gcli catalog --list", just keep one solution. Besides, you can support both specify "--catalog" and "--name", this will also introduce inconsistency if user both set these two arguments, why don't you just keep one solution like "--name", which should be enough.

Reply:
OK I'll remove those options. --name can be used to specify metalake.catalog.schma.table but not other entities like tag or role or group so there will need to be other options for them.

Reply:
Done